### PR TITLE
Declare assert.h

### PR DESCRIPTION
**Pull request by NeKroFR**

### CODE REVIEW

File: sources/libc/ansi/assert.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+NDEBUG
+
+#define assert(ignore) ((void)0)
+
+static_assert
+
+/* --- 7.2.1 Program diagnostics ----------------------------------------------- */ 
+
+void assert(scalar expression);
```
Comment (sleepy-monax) [inline sources/libc/ansi/assert.h:13]: This is not valid C

**Review by sleepy-monax — CHANGES_REQUESTED** [review]

File: sources/libc/ansi/assert.h (hunk 1 above)
Comment (NeKroFR) [reply sources/libc/ansi/assert.h:13]: ok sorry i will see how to change it

Comment (NeKroFR) [reply sources/libc/ansi/assert.h]: I add some changes, it is ok now @sleepy-monax ?

File: sources/libc/ansi/assert.h [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+/* --- 7.2 Diagnostics ----------------------------------------------- */ 
+
+void NDEBUG;
+
+void static_assert;
```
Comment (Supercip971) [inline sources/libc/ansi/assert.h:7]: static assert should be a macro that link to the _Static_assert builtin function

File: sources/libc/ansi/assert.h [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+/* --- 7.2 Diagnostics ----------------------------------------------- */ 
+
+void NDEBUG;
```
Comment (Supercip971) [inline sources/libc/ansi/assert.h:5]: NDEBUG should be a macro

Comment (Supercip971) [reply sources/libc/ansi/assert.h:5]: the macro NDEBUG is not defined in <assert.h>

File: sources/libc/ansi/assert.h [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+void static_assert;
+
+/* --- 7.2.1 Program diagnostics ----------------------------------------------- */ 
+
+void assert(scalar expression);
```
Comment (Supercip971) [inline sources/libc/ansi/assert.h:11]: assert should be a macro

Comment (Supercip971) [reply sources/libc/ansi/assert.h:11]: and should be ignored when NDEBUG is defined

**Review by Supercip971 — CHANGES_REQUESTED** [review]

File: sources/libc/ansi/assert.h [hunk 5]
```diff
@@ -1,11 +1,16 @@
 #pragma once
 
-/* --- 7.2 Diagnostics ----------------------------------------------- */ 
+/* ---define assert in terms of NDEBUG ----------------------------------------------- */ 
 
-void NDEBUG;
+#ifdef NDEBUG
+#define asser(ignore) ((void)0)
```
Comment (sleepy-monax) [inline sources/libc/ansi/assert.h:6]: assert

Comment (NeKroFR) [reply sources/libc/ansi/assert.h]: I fixed my stupidity

**Pull request merged**